src/window.rs

```diff
@@ -92,23 +92,19 @@ impl WindowId {
     pub const fn dummy() -> Self {
         WindowId(platform_impl::WindowId::dummy())
     }
-}
 
-impl fmt::Debug for WindowId {
-    fn fmt(&self, fmtr: &mut fmt::Formatter<'_>) -> fmt::Result {
-        self.0.fmt(fmtr)
+    pub fn into_raw(self) -> u64 {
```

Comment:
Now that we're making this change, would it make sense to use `*const ()` instead of `u64`? That'd carry the provenance of the ID, which, to be fair, it isn't actually important for the comparisons that we're making, but might allow this to be `const` (I think?).

But maybe I'm thinking too much of removing `WindowId` once we have `WindowHandle`, because I think it would be a requirement that we use a pointer there.

Reply:
It might indeed be worth a discussion removing `WindowId` entirely, but the use-cases mentioned above have to be covered somehow. (I still didn't understand the use-cases above, a matter for another time)

Reply:
> Nit/thought: Now that we're making this change, would it make sense to use `*const ()` instead of `u64`?

I forgot to answer this particular part:

- Android: implements no `WindowId` to being with.
- iOS: is a pointer.
- MacOS: is a pointer but is converted to a `u64` before being stored in `WindowId`.
- Orbital: is actually a pointer but is converted to a `u64` before being stored in `WindowId`.
- Wayland: is a pointer but is converted to a `u64` before being stored in `WindowId`.
- Web: implements a counter as a `WindowId`, but could use a pointer to the canvas like in `RawWindowHandle`.
- X11: is an integer.

So as far as I can tell X11 is the only problem.
We could simply return a pointer as well and document that it is just an integer in reality.

I guess we can do that in a follow-up.

Reply:
Yeah, I was kinda rambling here